app/src/main/java/com/framgia/soundclound/data/source/local/sqlite/AlbumLocalDataSource.java

```diff
@@ -0,0 +1,262 @@
+package com.framgia.soundclound.data.source.local.sqlite;
+
+import android.content.ContentValues;
+import android.content.Context;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteOpenHelper;
+
+import com.framgia.soundclound.data.model.Album;
+import com.framgia.soundclound.data.model.Track;
+import com.framgia.soundclound.data.source.AlbumDataSource;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.framgia.soundclound.data.model.Album.AlbumEntity.ID_ALBUM;
+import static com.framgia.soundclound.data.model.Album.AlbumEntity.IMAGE_ALBUM;
+import static com.framgia.soundclound.data.model.Album.AlbumEntity.LIST_TRACK;
+import static com.framgia.soundclound.data.model.Album.AlbumEntity.NAME_ALBUM;
+import static com.framgia.soundclound.data.model.Album.AlbumEntity.NUMBER_SONG;
+import static com.framgia.soundclound.util.Constant.DATABASE_NAME;
+import static com.framgia.soundclound.util.Constant.TABLE_ALBUM;
+
+
+/**
+ * Created by Bui Danh Nam on 8/1/2018.
+ */
+
+public class AlbumLocalDataSource extends SQLiteOpenHelper implements AlbumDataSource {
+
+    public static int VERSION = 1;
```

Comment:
ERROR: [MALICIOUS_CODE][MS_SHOULD_BE_FINAL] com.framgia.soundclound.data.source.local.sqlite.AlbumLocalDataSource.VERSION isn't final but should be
http://findbugs.sourceforge.net/bugDescriptions.html#MS_SHOULD_BE_FINAL